src/views/Stack/Card.tsx

```diff
@@ -0,0 +1,518 @@
+  // Changing it during an animations can result in unexpected results
+  private getInterpolatedStyle = memoize(
+    (
+      styleInterpolator: CardStyleInterpolator,
+      current: Animated.Node<number>,
+      next: Animated.Node<number> | undefined,
+      layout: Layout
+    ) =>
+      styleInterpolator({
+        progress: {
+          current,
+          next,
+        },
+        closing: this.isClosing,
+        layouts: {
+          screen: layout,
+        },
+      })
+  );
+
+  private gestureActivationCriteria() {
+    const { layout, direction, gestureResponseDistance } = this.props;
+
+    // Doesn't make sense for a response distance of 0, so this works fine
+    const distance =
+      direction === 'vertical'
+        ? (gestureResponseDistance && gestureResponseDistance.vertical) ||
+          GESTURE_RESPONSE_DISTANCE_VERTICAL
+        : (gestureResponseDistance && gestureResponseDistance.horizontal) ||
+          GESTURE_RESPONSE_DISTANCE_HORIZONTAL;
+
+    if (direction === 'vertical') {
+      return {
+        maxDeltaX: 15,
+        minOffsetY: 5,
+        hitSlop: { bottom: -layout.height + distance },
+      };
+    } else {
+      return {
+        minOffsetX: 5,
+        maxDeltaY: 20,
+        hitSlop: { right: -layout.width + distance },
+      };
+    }
+  }
+
+  private gestureRef: React.Ref<PanGestureHandler> = React.createRef();
+
+  render() {
+    const {
+      active,
+      transparent,
+      layout,
+      current,
+      next,
+      direction,
+      overlayEnabled,
+      shadowEnabled,
+      gesturesEnabled,
+      children,
+      styleInterpolator,
+      ...rest
+    } = this.props;
+
+    const {
+      containerStyle,
+      cardStyle,
+      overlayStyle,
+      shadowStyle,
+    } = this.getInterpolatedStyle(styleInterpolator, current, next, layout);
+
+    const handleGestureEvent =
+      direction === 'vertical'
+        ? this.handleGestureEventVertical
+        : this.handleGestureEventHorizontal;
+
+    return (
+      <StackGestureContext.Provider value={this.gestureRef}>
+        <View pointerEvents="box-none" {...rest}>
+          <Animated.Code exec={this.exec} />
+          {overlayEnabled && overlayStyle ? (
+            <Animated.View
+              pointerEvents="none"
+              style={[styles.overlay, overlayStyle]}
+            />
+          ) : null}
+          <Animated.View
+            style={[styles.container, containerStyle]}
+            pointerEvents="box-none"
+          >
+            <PanGestureHandler
+              ref={this.gestureRef}
+              enabled={layout.width !== 0 && gesturesEnabled}
+              onGestureEvent={handleGestureEvent}
+              onHandlerStateChange={handleGestureEvent}
+              {...this.gestureActivationCriteria()}
+            >
+              <Animated.View style={[StyleSheet.absoluteFill, cardStyle]}>
+                {shadowEnabled && !transparent ? (
```

Comment:
Maybe just a truly minor thing, but sometimes you use `!x ? val : null` and somewhere `Boolean(x) && val`

Reply:
haha I usually prefer the ternary. maybe the boolean stuff was leftover when copying some code from old code